vpr/src/route/route_breadth_first.cpp

```diff
@@ -384,15 +384,16 @@ static void breadth_first_expand_neighbours(BinaryHeap& heap, int inode, float p
      * heap.  pcost is the path_cost to get to inode.                           */
 
     auto& device_ctx = g_vpr_ctx.device();
+    const auto& rr_graph = device_ctx.rr_graph;
     auto& route_ctx = g_vpr_ctx.routing();
 
     for (RREdgeId from_edge : device_ctx.rr_nodes.edge_range(RRNodeId(inode))) {
         RRNodeId to_node = device_ctx.rr_nodes.edge_sink_node(from_edge);
 
-        if (device_ctx.rr_nodes.node_xhigh(to_node) < route_ctx.route_bb[net_id].xmin
-            || device_ctx.rr_nodes.node_xlow(to_node) > route_ctx.route_bb[net_id].xmax
-            || device_ctx.rr_nodes.node_yhigh(to_node) < route_ctx.route_bb[net_id].ymin
-            || device_ctx.rr_nodes.node_ylow(to_node) > route_ctx.route_bb[net_id].ymax)
+        if (rr_graph.node_xhigh(to_node) < route_ctx.route_bb[net_id].xmin
```

Comment:
This may be an API we want to. Given a bounding box (you can refer to ``vtr::Rect<short>``), we check if the coordinates of a routing resource node is inside or not. This may be used by routers frequently.